modules/internal/Bytes.chpl

```diff
@@ -677,405 +676,406 @@ module Bytes {
     for s in doSplitWSNoEnc(this, maxsplit) do yield s;
   }
 
-    /*
-      Returns a new :mod:`bytes <Bytes>`, which is the concatenation of all of
-      the :mod:`bytes <Bytes>` passed in with the contents of the method
-      receiver inserted between them.
+  /*
+    Returns a new :mod:`bytes <Bytes>`, which is the concatenation of all of
+    the :mod:`bytes <Bytes>` passed in with the contents of the method
+    receiver inserted between them.
 
-      .. code-block:: chapel
+    .. code-block:: chapel
 
-          var myBytes = b"|".join(b"a",b"10",b"d");
-          writeln(myBytes); // prints: "a|10|d"
+        var myBytes = b"|".join(b"a",b"10",b"d");
+        writeln(myBytes); // prints: "a|10|d"
 
-      :arg x: :mod:`bytes <Bytes>` values to be joined
+    :arg x: :mod:`bytes <Bytes>` values to be joined
 
-      :returns: A :mod:`bytes <Bytes>`
-    */
-    inline proc bytes.join(const ref x: bytes ...) : bytes {
-      return _join(x);
-    }
+    :returns: A :mod:`bytes <Bytes>`
+  */
+  inline proc bytes.join(const ref x: bytes ...) : bytes {
+    return doJoin(this, x);
+  }
 
-    /*
-      Returns a new :mod:`bytes <Bytes>`, which is the concatenation of all of
-      the :mod:`bytes <Bytes>` passed in with the contents of the method
-      receiver inserted between them.
+  /*
+    Returns a new :mod:`bytes <Bytes>`, which is the concatenation of all of
+    the :mod:`bytes <Bytes>` passed in with the contents of the method
+    receiver inserted between them.
 
-      .. code-block:: chapel
+    .. code-block:: chapel
 
-          var tup = (b"a",b"10",b"d");
-          var myBytes = b"|".join(tup);
-          writeln(myBytes); // prints: "a|10|d"
+        var tup = (b"a",b"10",b"d");
+        var myJoinedTuple = b"|".join(tup);
+        writeln(myJoinedTuple); // prints: "a|10|d"
 
-      :arg x: :mod:`bytes <Bytes>` values to be joined
-      :type x: tuple or array of :mod:`bytes <Bytes>`
+        var myJoinedArray = b"|".join([b"a",b"10",b"d"]);
+        writeln(myJoinedArray); // prints: "a|10|d"
 
-      :returns: A :mod:`bytes <Bytes>`
-    */
-    inline proc bytes.join(const ref x) : bytes where isTuple(x) {
-      if !isHomogeneousTuple(x) || !isBytes(x[1]) then
-        compilerError("join() on tuples only handles homogeneous tuples of bytes");
-      return _join(x);
-    }
+    :arg x: An array or tuple of :mod:`bytes <Bytes>` values to be joined
 
-    /*
-      Strips given set of leading and/or trailing characters.
+    :returns: A :mod:`bytes <Bytes>`
+  */
+  inline proc bytes.join(const ref x) : bytes  {
+    compilerError("bytes.join() accepts any number of bytes, homogenous "
```

Comment:
Seeing the code, I wonder if implementing the function here would make more sense from a maintainability standpoint. But I'll leave that to you. I don't have a strong preference.

If we keep things as-is, could you throw a comment here saying something along the lines of "join is handled by the primary methods above"?

Reply:
I added the comments to make it more clear that the logic for the overloads is elsewhere.